### PR TITLE
changed: get rid of find modules for opm modules

### PR DESCRIPTION
**Pull request by akva2**
use pure config mode.

we no longer compile a test application for the libraries,
however checks have been tightened by making sure version of
all opm modules are the same.

this allows for putting prereqs directly in the module repositories

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams opm-upscaling=234 opm-simulators=1374 ewoms=256 opm-core=1201 opm-grid=294 opm-output=226 opm-material=270 opm-parser=1173 please

**Comment by akva2** [comment]
jenkins build this with downstreams opm-upscaling=234 opm-simulators=1374 ewoms=256 opm-core=1201 opm-grid=294 opm-output=226 opm-material=270 opm-parser=1173 please

**Comment by akva2** [comment]
jenkins build this with downstreams opm-upscaling=234 opm-simulators=1374 ewoms=256 opm-core=1201 opm-grid=294 opm-output=226 opm-material=270 opm-parser=1173 please

File: cmake/Modules/OpmFind.cmake [hunk 1]
```diff
@@ -147,36 +149,58 @@ macro (find_and_append_package_to prefix name)
...
-	endif ()
-	if (NOT DEFINED ${NAME}_FOUND)
-	  set (${NAME}_FOUND "${${name}_FOUND}")
-	endif ()
-
-       if(name STREQUAL "opm-common")
-         set(_opm_common_deps_processed ON)
-       endif()
+        find_package (${name} ${ARGN_NO_REQUIRED} NO_CMAKE_SYSTEM_PACKAGE_REGISTRY NO_CMAKE_PACKAGE_REGISTRY)
+        # If everything else failed fall back to the registry
```
Comment (blattms) [inline cmake/Modules/OpmFind.cmake]: Is this comment coorect? Seems like we just get some properties ecl-config.cmake defines for the target ecl and use them as variables.

Comment (akva2) [reply cmake/Modules/OpmFind.cmake]: nope, it should be removed. it clawed itself stuck and refused to go during reorganizations it seems.

**Comment by akva2** [comment]
builds on jenkins using installed builds (result gone due to removing a comment but it was all green). i have tested both sibling and dc builds locally. my co-maintainer for the buildsystem approves. unless somebody yells within the end of the day i will merge this series.

**Comment by atgeirr** [comment]
Thanks a lot! Not tested locally yet, but I would not want to delay this anyway... (will tell you if I have trouble of course)

**Pull request merged**